sktime/experiments/analysis.py

```diff
@@ -0,0 +1,374 @@
+
+import pandas as pd
+import numpy as np
+import itertools
+
+from scipy import stats
+from scipy.stats import ttest_ind
+from scipy.stats import ranksums
+
+from sklearn.metrics import accuracy_score, mean_squared_error
+import scikit_posthocs as sp
+
+from sktime.experiments.losses import Losses
+
+import matplotlib.pyplot as plt
+
+class AnalyseResults(object):
+    """
+    Analyze results of machine learning experiments.
+
+    Parameters
+    ----------
+    result: sktime result object
+        class for storing the results
+    """
+
+    def __init__(self, 
+                 results):
+
+        self._results_list = results.load()
```

Comment:
instead of calling `load` here, we could call it in line 47 when you iterate over results
```suggestion
        self._results_list = results
```

Reply:
looks kind of similar to me, is there a performance or other gain?

Reply:
You wouldn't load all of the results (true and predicted labels) into memory at the same time

Reply:
Your suggestion also loads everything into memory at the same time it just does it in a different place. 

The memory requirements for loading all the results should not be very big, I would assume.

Reply:
Not if the `load` method is an iterator ...